augur/util_support/node_data_file.py

```diff
@@ -63,49 +63,49 @@ def validate(self):
+                raise AugurError(
+                    f"`nodes` value in {self.fname} is not a dictionary. Please check the formatting of this JSON!"
                 )
-            except ValidateError as err:
+
+            if not isinstance(self.branches, dict):
                 raise AugurError(
-                    f"{self.fname} contains an `annotations` attribute of an invalid JSON format. Was it "
-                    "produced by different version of augur the one you are currently using "
-                    f" ({__version__})? Please check the program that produced that JSON file."
-                ) from err
-
-        if not isinstance(self.nodes, dict):
-            raise AugurError(
-                f"`nodes` value in {self.fname} is not a dictionary. Please check the formatting of this JSON!"
-            )
-
-        if not isinstance(self.branches, dict):
-            raise AugurError(
-                f"`branches` value in {self.fname} is not a dictionary. Please check the formatting of this JSON!"
-            )
-
-        if not self.nodes and not self.branches:
-            print_err(
-                f"WARNING: {self.fname} has empty or nonexistent `nodes` and `branches`. Please check the formatting of this JSON!"
-            )
-
-        if self.is_generated_by_incompatible_augur:
-            msg = (
-                f"Augur version incompatibility detected: the JSON {self.fname} was generated by "
-                f"{self.generated_by}, which is incompatible with the current augur version "
-                f"({__version__}). We suggest you rerun the pipeline using the current version of "
-                "augur."
-            )
-            if self.validation_mode is ValidationMode.ERROR:
+                    f"`branches` value in {self.fname} is not a dictionary. Please check the formatting of this JSON!"
+                )
+
+            if not self.nodes and not self.branches:
+                print_err(
+                    f"WARNING: {self.fname} has empty or nonexistent `nodes` and `branches`. Please check the formatting of this JSON!"
+                )
+
+            if self.is_generated_by_incompatible_augur:
+                msg = (
+                    f"Augur version incompatibility detected: the JSON {self.fname} was generated by "
+                    f"{self.generated_by}, which is incompatible with the current augur version "
+                    f"({__version__}). We suggest you rerun the pipeline using the current version of "
+                    "augur."
+                )
                 raise AugurError(msg)
-            elif self.validation_mode is ValidationMode.WARN:
-                print_err(f"WARNING: {msg}")
+        except AugurError as e:
+            if self.validation_mode is ValidationMode.WARN:
+                print_err(f"WARNING: {e.args[0]}")
             else:
-                raise ValueError(f"unknown validation mode: {self.validation_mode!r}")
-
+                raise e
```

Comment:
This can potentially catch _any_ `AugurError` that is raised during validation, not just the errors explicitly raised in `validate`. Is that the desired behavior? 

E.g. within `load_json_schema` there's potential for [`open_file` to raise an AugurError](https://github.com/nextstrain/augur/blob/75fc490226f5ae3c0ecd81b67e8a4b8ceb950961/augur/io/file.py#L49). This would then warn that the schema file is not valid without doing any validation that outputs as warnings. 

--- 

We can change all of the `AugurError`s within `validate` to `ValidateError` and only catch `ValidateError` here.

Reply:
👍 [here's the diff](https://github.com/nextstrain/augur/compare/1a9cc7f32dfc62601e7287c53635a0bd932f15f5..a43536c2fa795ec00007923bb9877596413e6607) implementing this suggestion